File: rb/features.rb
# IpGeolocationApi2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module IpGeolocationApi2Features
  def self.make_feature(name)
    case name
    when "base"
      IpGeolocationApi2BaseFeature.new
    when "ratelimit"
      IpGeolocationApi2RatelimitFeature.new
    when "retry"
      IpGeolocationApi2RetryFeature.new
    when "test"
      IpGeolocationApi2TestFeature.new
    when "timeout"
      IpGeolocationApi2TimeoutFeature.new
    else
      IpGeolocationApi2BaseFeature.new
    end
  end
end

File: py/features.py
# IpGeolocationApi2 SDK feature factory

from feature.base_feature import IpGeolocationApi2BaseFeature
from feature.test_feature import IpGeolocationApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpGeolocationApi2BaseFeature(),
        "test": lambda: IpGeolocationApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
